### PR TITLE
Updated build environment documentation

### PR DESCRIPTION
**Pull request by matsduf**
* Updated BuildEnvironmentPreparation.md from comments in PR #894.
* Removed some trailing spaces.
* Added link to BuildEnvironmentPreparation.md
  from ReleaseProcess-create-test-distribution.md

### CODE REVIEW

File: docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md [hunk 1]
```diff
@@ -5,7 +5,13 @@ The purpose of this part is to create the test distributions that can be
 used in QA testing for a release. It can also be used to create test
 distributions to verify a pull request.
 
-## 1. Create a clean environment
+
+## 1. Prepare a build environment
+Set up build system to be used for the test distribution creation. See
+[Build Environment Preparation] for set it up.
```
Comment (pnax) [inline docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md:11]: `for set it up` -> I would say `for set up` or `to set it up` (but I'm not 100% sure)

File: docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md [hunk 2]
```diff
@@ -69,7 +75,7 @@ MANIFEST.SKIP, i.e. no missing or extra files:
     make distcheck
 
 
-## 4. Produce distribution tarballs
+## 5. Produce distribution tarballs
```
Comment (pnax) [inline docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md:78]: There are two `5` sections (this looks odd) :
`## 5. Produce distribution tarballs`
`## 5. Produce distribution zip file`

Maybe the second one should be a `5.1` :
 `### 5.1. Produce distribution zip file`

Comment (matsduf) [reply docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md:78]: It should then be 5 and 6 respectively. I will correct.

File: docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md (hunk 1 above)
Comment (matsduf) [reply docs/internal-documentation/maintenance/ReleaseProcess-create-test-distribution.md:11]: Fixed.

**Review by pnax — APPROVED** [review]

**Review by mattias-p — APPROVED** [review]

**Pull request merged**